botorch/optim/initializers.py

```diff
@@ -226,6 +229,130 @@ def gen_one_shot_kg_initial_conditions(
+
+    Args:
+        acq_function: The value function instance to be optimized.
+        bounds: A `2 x d` tensor of lower and upper bounds for each column of
+            task features.
+        num_restarts: The number of starting points for multistart acquisition
+            function optimization.
+        raw_samples: The number of raw samples to consider in the initialization
+            heuristic.
+        current_model: The model of the KG acquisition function that was used to
+            generate the fantasy model of the value function.
+        options: Options for initial condition generation. These contain all
+            settings for the standard heuristic initialization from
+            `gen_batch_initial_conditions`. In addition, they contain
+            `frac_random` (the fraction of fully random fantasy points),
+            `num_inner_restarts` and `raw_inner_samples` (the number of random
+            restarts and raw samples for solving the posterior objective
+            maximization problem, respectively) and `eta` (temperature parameter
+            for sampling heuristic from posterior objective maximizers).
+
+    Returns:
+        A `num_restarts x batch_shape x q x d` tensor that can be used as initial
+        conditions for `optimize_acqf()`. Here `batch_shape` is the
+        `_input_batch_shape` of value function model.
+
+    Example:
+        >>> fant_X = torch.rand(5, 1, 2)
+        >>> fantasy_model = model.fantasize(fant_X, SobolQMCNormalSampler(16))
+        >>> value_function = PosteriorMean(fantasy_model)
+        >>> bounds = torch.tensor([[0., 0.], [1., 1.]])
+        >>> Xinit = gen_value_function_initial_conditions(
+        >>>     value_function, bounds, num_restarts=10, raw_samples=512,
+        >>>     options={"frac_random": 0.25},
+        >>> )
+    """
+    # TODO: should this be added to __init__?
+    options = options or {}
+    seed: Optional[int] = options.get("seed")
+    batch_limit: Optional[int] = options.get("batch_limit")
+    frac_random: float = options.get("frac_random", 0.2)  # TODO: increase?
+    if not 0 < frac_random < 1:
+        raise ValueError(
+            f"frac_random must take on values in (0,1). Value: {frac_random}"
+        )
+
+    # compute maximizer of the current value function
+    value_function = _get_value_function(
+        model=current_model,
+        objective=acq_function.objective,
+        sampler=acq_function.sampler
+        if isinstance(acq_function, qSimpleRegret)
+        else None,
+    )
+    from botorch.optim.optimize import optimize_acqf
+
+    fantasy_cands, fantasy_vals = optimize_acqf(
+        acq_function=value_function,
+        bounds=bounds,
+        q=1,
+        # TODO: set these to num_restarts and raw_samples?
+        num_restarts=options.get("num_inner_restarts", 20),
+        raw_samples=options.get("raw_inner_samples", 1024),
+        return_best_only=False,
```

Comment:
probably want to pass through `batch_limit` here

Reply:
passed the full `options` dict to be future compatible